airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -91,10 +92,12 @@ def read(
         logger: logging.Logger,
         config: Mapping[str, Any],
         catalog: ConfiguredAirbyteCatalog,
-        state: MutableMapping[str, Any] = None,
+        state: Union[List[AirbyteStateMessage], MutableMapping[str, Any]] = None,
```

Comment:
We should never expect a `MutableMapping[str, Any]`, but in order to retain backwards compatibility with OSS users who pull the latest CDK, we should continue to support the type as defined in the interface. The connector manager under the hood knows how to interpret and store this as legacy.